IBPSA/Fluid/BaseClasses/MassFlowRateMultiplier.mo

```diff
@@ -2,17 +2,55 @@ within IBPSA.Fluid.BaseClasses;
 model MassFlowRateMultiplier "Model that multiplies the mass flow rate"
   extends IBPSA.Fluid.Interfaces.PartialTwoPort;
 
-  parameter Real k "Gain for mass flow rate";
+  parameter Boolean use_input=false
+    "Set to true for multiplier factor provided as an input instead of a parameter"
+    annotation(Evaluate=true);
+  parameter Real k=1
+    "Gain for mass flow rate"
+    annotation(Dialog(enable=not use_input));
+
+  Modelica.Blocks.Interfaces.RealInput u(
+    final unit="1",
+    final min=Modelica.Constants.small) if use_input
+    "Multiplier factor"
+    annotation (Placement(transformation(extent={{-140,40},{-100,80}}),
+        iconTransformation(extent={{-140,40},{-100,80}})));
+  Modelica.Blocks.Interfaces.RealOutput uInv if use_input
+    "Inverse of the multiplier factor"
+    annotation (Placement(transformation(
+          extent={{100,50},{120,70}}), iconTransformation(extent={{100,50},{120,
+            70}})));
+protected
+  Modelica.Blocks.Interfaces.RealInput u_internal
+    "Connector for multiplier factor for internal use only"
+    annotation (Placement(
+    transformation(extent={{-60,20},{-20,60}}),iconTransformation(extent={{100,40},
+    {140,80}})));
+
+  Modelica.Blocks.Sources.Constant cst(
+    final k=k) if not use_input
+    "Constant gain value"
+    annotation (Placement(transformation(extent={{-90,30},{-70,50}})));
+  Modelica.Blocks.Math.Division div1 if use_input
+    "Compute the inverse"
+    annotation (Placement(transformation(extent={{20,50},{40,70}})));
+  Modelica.Blocks.Sources.Constant one(
+    final k=1) if use_input
+    "Constant 1"
+    annotation (Placement(transformation(extent={{-40,70},{-20,90}})));
 
 initial equation
-  assert(k > Modelica.Constants.small or -k < -Modelica.Constants.small,
```

Comment:
I changed this by removing the 2nd condition which is not needed, and updated the documentation to state that the gain must be strictly positive.

Reply:
Looks good to me.